### DISCUSSION TITLE
How to resolve Unsatisfied dependency exception

### DISCUSSION THREAD

**Discussion by abhishekDautomator** [quarkusio/quarkus#32344] [Community]
I am new to the Quarkus framework and I am stuck with the below error. Any suggestion would be very helpful on where is the issue.

**Error:**
`[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.abhishek.dao.PersonRepository and qualifiers [@Default] - java member: org.abhishek.resource.PersonResource#repository - declared on CLASS bean [types=[org.abhishek.resource.PersonResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.abhishek.resource.PersonResource]`

**Entity:**
`@Entity
@Table(name = "persons")
public class PersonEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "age")
    private int age;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}`

**Repository:**
`public interface PersonRepository extends PanacheRepository<PersonEntity> {
}`

**Resource:**
`@ApplicationScoped
@Path("/persons")
public class PersonResource {

    @Inject
    PersonRepository repository;

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public List<PersonEntity> list() {
        return repository.listAll();
    }

    @POST
    @Consumes(MediaType.APPLICATION_JSON)
    public void add(PersonEntity person) {
        repository.persist(person);
    }
}`

I am trying to run my application and connect to the Postgres DB I am running locally on port 5432. The application should be able to connect to the DB and create persons table and I would be able to hit the APIs configured. But getting the above error, which I am not able to resolve.

**Comment by luca-bassoricci** [comment]
https://stackoverflow.com/questions/75915951/getting-javax-enterprise-inject-unsatisfiedresolutionexception-when-running-a-qu

**Reply by mkouba** [reply]
For the record, a [Panache repository](https://quarkus.io/guides/hibernate-orm-panache#defining-your-repository) must be a CDI bean; typically a class annotated with `@ApplicationScoped`. I.e. `interface PersonRepository extends PanacheRepository<PersonEntity>` is just ignored.

**Comment by gopalnaik161** [comment]
Iam getting unsatisfied dependency injection when i make encrypted war and run it .. but when iam executing in the sts iaam not getting this issue. Java 8 and using extentreport dependency..
